nexus/src/external_api/console_api.rs

```diff
@@ -387,17 +401,34 @@ pub struct LoginPathParam {
     pub silo_name: crate::db::model::Name,
 }
 
+#[endpoint {
+   method = GET,
+   path = "/login/{silo_name}/local",
+   tags = ["login"],
+   unpublished = true,
+}]
+pub async fn login_local_begin(
+    rqctx: RequestContext<Arc<ServerContext>>,
+    _path_params: Path<LoginPathParam>,
+) -> Result<Response<Body>, HttpError> {
+    // TODO: figure out why instrumenting doesn't work
+    // let apictx = rqctx.context();
+    // let handler = async { serve_console_index(rqctx.context()).await };
+    // apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
+    serve_console_index(rqctx.context()).await
+}
+
 /// Authenticate a user via username and password
 #[endpoint {
    method = POST,
-   path = "/login/{silo_name}/local",
+   path = "/v1/login/{silo_name}/local",
```

Comment:
I intentionally left the `/v1/` off the console endpoints which the thought being that they'd need to be relatively stable. I can see wanting to gate the POST endpoint though given that it'll mostly be treated like any other API endpoint.

Reply:
Yeah, I went back and forth. It is a JSON API endpoint, so I figured it should probably follow the convention. I'm thinking we should get to a point where only the GETs don't have v1 and they're all unpublished. It feels weird to have prefixed and non-prefixed endpoints right next to each other, but that's where I landed as least unreasonable.

Reply:
I'm not sure. We could have `POST /v1/login` just always be username/password, and that would make this endpoint unnecessary. Or maybe we could have `POST /v1/login/local` and we'd change `POST /login/{silo_name}/saml/{provider_name}` to `POST /v1/login/saml/{provider_name}`?

Reply:
The latter part feels better than me. Given that username/password log ins should be relatively rare I don't know that I'd want it to be the root endpoint (which makes it feel like the default).

Reply:
Yeah, I think that's right.